File: scripts/generate-favicon.mjs
import sharp from "sharp";
import fs from "fs";
import path from "path";

const size = 512;
const MOLT_BLACK = "#070501";
const MOLT_WHITE = "#f7f1e7";

// Read the font file and base64-encode it for embedding in SVG
const projectRoot = "/vercel/share/v0-project";
const fontPath = path.join(projectRoot, "public", "fonts", "Moulay-Bold.ttf");
const fontBuffer = fs.readFileSync(fontPath);
const fontBase64 = fontBuffer.toString("base64");

const svg = `
<svg width="${size}" height="${size}" xmlns="http://www.w3.org/2000/svg">
  <defs>
    <style>
      @font-face {
        font-family: 'MoulayBold';
        src: url('data:font/truetype;base64,${fontBase64}');
        font-weight: bold;
        font-style: normal;
      }
    </style>
  </defs>
  <rect width="${size}" height="${size}" fill="${MOLT_BLACK}" />
  <text
    x="50%"
    y="50%"
    dominant-baseline="central"
    text-anchor="middle"
    font-family="MoulayBold"
    font-weight="bold"
    font-size="380"
    fill="${MOLT_WHITE}"
  >M</text>
</svg>
`;

const outputPath = path.join(projectRoot, "app", "icon.png");

sharp(Buffer.from(svg))
  .resize(512, 512)
  .png()
  .toFile(outputPath)
  .then(() => console.log("Favicon saved to", outputPath))
  .catch((err) => console.error("Error:", err));

File: .vscode/scripts/generate-favicon.py
from PIL import Image, ImageDraw, ImageFont

# MOLT brand colours
MOLT_BLACK = (7, 5, 1)
MOLT_WHITE = (247, 241, 231)

# Create 512x512 square canvas
size = 512
img = Image.new("RGB", (size, size), MOLT_BLACK)
draw = ImageDraw.Draw(img)

# Load Moulay-Bold font at a large size
import os
project_root = "/vercel/share/v0-project"
font_path = os.path.join(project_root, "public", "fonts", "Moulay-Bold.ttf")
font = ImageFont.truetype(font_path, 360)

# Measure the "M" glyph
text = "M"
bbox = draw.textbbox((0, 0), text, font=font)
text_width = bbox[2] - bbox[0]
text_height = bbox[3] - bbox[1]

# Centre the glyph on the canvas
x = (size - text_width) / 2 - bbox[0]
y = (size - text_height) / 2 - bbox[1]

draw.text((x, y), text, fill=MOLT_WHITE, font=font)

# Save as PNG (Next.js app/icon.png is auto-detected as favicon)
output_path = os.path.join(project_root, "app", "icon.png")
img.save(output_path, "PNG")
print(f"Favicon saved to {output_path}")
